src/back/spv/writer.rs

```diff
@@ -103,30 +96,36 @@ impl Writer {
         }
     }
 
-    fn get_type_id(
-        &mut self,
-        arena: &crate::Arena<crate::Type>,
-        handle: crate::Handle<crate::Type>,
-    ) -> Word {
-        match self.lookup_type.lookup_id(handle) {
-            Some(word) => word,
-            None => {
-                let (instruction, id) = self.parse_type_declaration(arena, handle);
-                instruction.to_words(&mut self.logical_layout.declarations);
-                id
+    fn get_type_id(&mut self, arena: &crate::Arena<crate::Type>, lookup_ty: LookupType) -> Word {
+        let mut sink: Vec<(LookupType, Word)> = vec![];
+
+        let id = match self.lookup_type.entry(lookup_ty) {
+            Entry::Occupied(e) => *e.get(),
+            Entry::Vacant(e) => {
+                let id = match lookup_ty {
+                    LookupType::Handle(handle) => self.write_type_declaration_arena(arena, handle, &mut sink),
+                    LookupType::Local(local_ty) => self.write_type_declaration_local(arena, local_ty),
+                };
+                *e.insert(id)
             }
+        };
+
+        for (k, v) in sink.iter() {
+          self.lookup_type.insert(*k, *v);
```

Comment:
I think this works, but it is not clean, as the sink will always be initialized even though a LookupType is local.